src/OmniSharp.Cake/Extensions/ResponseExtensions.cs

```diff
@@ -121,6 +124,85 @@ public static Task<QuickFixResponse> TranslateAsync(this QuickFixResponse respon
             return response;
         }
 
+        public static async Task<CodeStructureResponse> TranslateAsync(this CodeStructureResponse response, OmniSharpWorkspace workspace, CodeStructureRequest request)
+        {
+            var zeroIndex = await LineIndexHelper.TranslateToGenerated(request.FileName, 0, workspace);
+            var elements = new List<CodeElement>();
+
+            foreach (var element in response.Elements)
+            {
+                if (element.Ranges.Values.Any(x => x.Start.Line < zeroIndex))
+                {
+                    continue;
+                }
+
+                var translatedElement = await element.TranslateAsync(workspace, request);
+
+                if (translatedElement.Ranges.Values.Any(x => x.Start.Line < 0))
+                {
+                    continue;
+                }
+
+                elements.Add(translatedElement);
+            }
+
+            response.Elements = elements;
+            return response;
+        }
+
+        private static async Task<CodeElement> TranslateAsync(this CodeElement element, OmniSharpWorkspace workspace, SimpleFileRequest request)
+        {
+            var builder = new CodeElement.Builder
+            {
+                Kind = element.Kind,
+                DisplayName = element.DisplayName,
+                Name = element.Name
+            };
+
+            foreach (var property in element.Properties ?? Enumerable.Empty<KeyValuePair<string, object>>())
+            {
+                builder.AddProperty(property.Key, property.Value);
+            }
+
+            foreach (var range in element.Ranges ?? Enumerable.Empty<KeyValuePair<string, Range>>())
+            {
+                builder.AddRange(range.Key, await range.Value.TranslateAsync(workspace, request));
+            }
+
+            foreach (var childElement in element.Children ?? Enumerable.Empty<CodeElement>())
+            {
+                var translatedElement = await childElement.TranslateAsync(workspace, request);
+
+                // This is plain stupid, but someone might put a #load directive inside a method or class
```

Comment:
is this supported in Cake?
In scripting this would generate `CS8098 "Cannot use #load after first token in file"`
https://github.com/dotnet/roslyn/blob/ca2f6ad9ab68518344c2d3e25c0731c83d136701/src/Compilers/CSharp/Portable/Errors/ErrorCode.cs#L1329

Reply:
Yes it is supported since Cake only includes the content of `#load`:ed files. It doesn't rely on the Roslyn `#load` directive currently. I'm about to look into changing that behavior however 😄